src/Component/Form/Type/VariationChoiceType.php

```diff
@@ -49,13 +49,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $choices = $this->pool->getProvider($options['product'])->getVariationsChoices($options['product'], $options['fields']);
 
         foreach ($choices as $choiceTitle => $choiceValues) {
+            $choiceOptions = array(
+                'label' => sprintf('form_%s', $choiceTitle),
+                'translation_domain' => 'SonataProductBundle',
+            );
+            // NEXT_MAJOR: Remove this "if" (when requirement of Symfony is >= 2.7)
+            if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
+                $choiceValues = array_flip($choiceValues);
+                // choice_as_value option is not needed in SF 3.0+
+                if (method_exists('Symfony\Component\Form\FormTypeInterface', 'setDefaultOptions')) {
+                    $choiceOptions['choices_as_values'] = true;
+                }
+            }
+            $choiceOptions['choices'] = $choiceValues;
+
             $builder->add($choiceTitle, $choiceType, array_merge(
-                    array('translation_domain' => 'SonataProductBundle'),
```

Comment:
This should remain as is, because otherwise you won't be able to override this `translation_domain` with your custom options. Am I wrong?

Not sure if you can use $options['field_options'] to pass a custom `translation_domain`.

It is only on this choice type tho, because the other didn't even had the option before.